### PR TITLE
8293540: [Metrics] Incorrectly detected resource limits with additional cgroup fs mounts

### PR DESCRIPTION
**Pull request by jerboaa**
Similar issue to the hotspot change discussed in https://bugs.openjdk.org/browse/JDK-8293472. The Java metrics implementation may get the resource limits wrong if there are additional cgroup fs mounts. Apparently that's more common than one might think. I've reproduced this with these existing tests on cg v2:

```
test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java
test/jdk/jdk/internal/platform/docker/TestDockerCpuMetrics.java
test/jdk/jdk/internal/platform/docker/TestDockerMemoryMetrics.java 
```

I've also added `test/jdk/jdk/internal/platform/docker/TestDockerBasic.java` and amended `test/jdk/jdk/internal/platform/cgroup/TestCgroupSubsystemFactory.java` which unconditionally fails (irrespective of cgroup version in use). The fix is fairly straight forward and is an extension which we already do for the `cpuset` controller: Allow duplicates, and if there are any prefer those mounted at `/sys/fs/cgroup`.

Testing:
- [x] fastdebug build on cgroups v2 and cgroups v1 (before and after the product fix)
- [x] added tests fail before, pass after the product fix.
- [x] Some manual testing using `cgcreate` and `cgexec` on cg1 and cg2. Still pass. 
- [x] GHA all pass.

Please review! Many thanks in advance.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293540](https://bugs.openjdk.org/browse/JDK-8293540): [Metrics] Incorrectly detected resource limits with additional cgroup fs mounts

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to [8059dd81](https://git.openjdk.org/jdk/pull/10248/files/8059dd8147bece81f8e35d2c75077ca1034a89c0)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10248/head:pull/10248` \
`$ git checkout pull/10248`

Update a local copy of the PR: \
`$ git checkout pull/10248` \
`$ git pull https://git.openjdk.org/jdk pull/10248/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10248`

View PR using the GUI difftool: \
`$ git pr show -t 10248`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10248.diff">https://git.openjdk.org/jdk/pull/10248.diff</a>

</details>

### CODE REVIEW

**Comment by dcubed-ojdk** [comment]
I fixed a typo in the bug's synopsis line so this PR's title needs to be update.
Doing a "/issue JDK-8293540" will do the trick.

**Comment by jerboaa** [comment]
Thanks! I've updated the PR title to match the bug synopsis.

**Comment by jerboaa** [comment]
Anyone willing to review this? Thanks!

File: test/jdk/jdk/internal/platform/docker/TestDockerBasic.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Verify that -XshowSettings:system works
```
Comment (iklam) [inline test/jdk/jdk/internal/platform/docker/TestDockerBasic.java:26]: Add `@bug 8293540`

File: test/jdk/jdk/internal/platform/docker/TestDockerBasic.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private static void testXshowSettingsSystem(boolean addCgroupMounts) throws Exception {
+        String testMsg = (addCgroupMounts ? " with " : " without ") + " additional cgroup FS mounts in /cgroup-in";
+        Common.logNewTestCase("Test TestDockerBasic " + testMsg);
+        DockerRunOptions opts =
+                new DockerRunOptions(imageName, "/jdk/bin/java", "-version");
+        opts.addJavaOpts("-esa");
+        opts.addJavaOpts("-XshowSettings:system");
+        opts.addDockerOpts("--memory", "300m");
+        if (addCgroupMounts) {
+            opts.addDockerOpts("--volume", "/sys/fs/cgroup:/cgroup-in:ro");
```
Comment (iklam) [inline test/jdk/jdk/internal/platform/docker/TestDockerBasic.java:64]: Add comments that the extra mount should be ignored by the cgroup set-up code. (Same for other test cases).

**Review by iklam — APPROVED** [review]
The JDK change looks good to me. Some nits for the test cases.

File: test/jdk/jdk/internal/platform/docker/TestDockerBasic.java (hunk 2 above)
Comment (jerboaa) [reply test/jdk/jdk/internal/platform/docker/TestDockerBasic.java:66]: OK. Added in the updated version.

File: test/jdk/jdk/internal/platform/docker/TestDockerBasic.java (hunk 1 above)
Comment (jerboaa) [reply test/jdk/jdk/internal/platform/docker/TestDockerBasic.java:27]: Thanks, added.

**Comment by jerboaa** [comment]
Thanks for the review @iklam!

**Pull request closed without merging**